apps/web/app/api/v1/users/me/route.ts

```diff
@@ -37,3 +37,113 @@ export async function PUT(request: NextRequest) {
 
   return NextResponse.json(user);
 }
+
+const deleteUser = async (userId: string) => {
+  await prisma.user.delete({
+    where: {
+      id: userId,
+    },
+  });
+};
+const deleteMembership = async (teamId: string, userId: string) => {
+  await prisma.membership.delete({
+    where: {
+      userId_teamId: {
+        userId,
+        teamId,
+      },
+    },
+  });
+};
+
+// I created this type because I don't have access to prisma unums.
+// TODO find a way to get access to prisma enums
+type MembershipRole = "admin" | "owner";
+
+const updateUserMembership = async (teamId: string, userId: string, role: MembershipRole) => {
+  await prisma.membership.update({
+    where: {
+      userId_teamId: {
+        userId,
+        teamId,
+      },
+    },
+    data: {
+      role,
+    },
+  });
+};
+
+const getAdminMemberships = (memberships) => memberships.filter((membership) => membership.role === "admin");
+
+const deleteTeam = async (teamId: string) => {
+  await prisma.team.delete({
+    where: {
+      id: teamId,
+    },
+  });
+};
+
+export async function DELETE() {
+  try {
+    const currentUser = await getSessionUser();
+
+    if (!currentUser) {
+      return new Response("Not authenticated", {
+        status: 401,
+      });
+    }
+
+    const currentUserMemberships = await prisma.membership.findMany({
+      where: {
+        userId: currentUser.id,
+      },
+      include: {
+        team: {
+          select: {
+            id: true,
+            name: true,
+            memberships: {
+              select: {
+                userId: true,
+                role: true,
+              },
+            },
+          },
+        },
+      },
+    });
+
+    for (const currentUserMembership of currentUserMemberships) {
+      const teamMemberships = currentUserMembership.team.memberships;
+      const role = currentUserMembership.role;
+      const teamId = currentUserMembership.teamId;
+
+      const teamAdminMemberships = getAdminMemberships(teamMemberships);
+      const teamHasAtLeastOneAdmin = teamAdminMemberships.length > 0;
+      const teamHasOnlyOneMember = teamMemberships.length === 1;
+      const currentUserIsTeamOwner = role === "owner";
+
+      if (teamHasOnlyOneMember) {
+        await deleteTeam(teamId);
+      } else {
+        if (currentUserIsTeamOwner) {
+          if (teamHasAtLeastOneAdmin) {
+            const firstAdmin = teamAdminMemberships[0];
+            await updateUserMembership(teamId, firstAdmin.userId, "owner");
+          } else {
+            await deleteMembership(teamId, currentUser.id);
```

Comment:
In this case, the person deleting the account is the owner of the team, and there are other members who are not admins (viewers, editors,..).
Just deleting the owner's membership and leaving the team without an owner and without an admin seems like a bad option. In this case, please delete the whole team and we will update the delete info text.